### PR TITLE
Functionality for pseudoterminals in linux sandbox

### PR DESCRIPTION
**Pull request by crydell-ericsson**
As brought up in issue #5373 , the Linux sandbox does not allow processes that run inside it to open pseudoterminals. These changes enable this by addressing the two main underlying issues:

- `/dev/pts` can not be read-only if a new pseudoterminal is to be created. These changes make `dev/pts` writable when remounting file systems during sandbox initialization.
- The group associated with pseudoterminals is "tty". After creating a new pseudoterminal, its gid has to be changed. If there is no gid mapping in the user namespace that corresponds to "tty", this group will not be known inside the sandbox. This causes issues in some Linux distributions, since they do not allow changing the group of a file to one that is not known inside the current user namespace. These changes map the gid of the user to the one corresponding to "tty" inside the sandbox in order to avoid this issue.

These changes introduce the `-P` flag to `linux-sandbox` in order to control whether or not the changes are applied, and the `--sandbox-explicit-pseudoterminal` to `bazel` in order to set this when calling bazel.

### CODE REVIEW

**Comment by philwo** [comment]
Thank you @crydell-ericsson! Ideally I'd like to avoid adding yet another flag... is there a way to automatically detect what the right thing to do is?

Could we just always mount `/dev/pts` and somehow map or make the `tty` group usable inside the sandbox without explicitly changing to it?

**Comment by crydell-ericsson** [comment]
Just thought I'd check in on the progress of this pull request; is there any additional information needed for this?

**Comment by philwo** [comment]
Thank you for the information, @crydell-ericsson. It's too bad that the best solution requires CAP_SETGID. 😞 I agree with your thoughts.

@larsrc-google Let's merge this as is? Looks clean to me and all changed behavior is behind a flag.

File: src/main/tools/linux-sandbox-pid1.cc [hunk 1]
```diff
@@ -232,8 +233,9 @@ static void SetupUserNamespace() {
     inner_gid = global_outer_gid;
   }
   if (opt.enable_pty) {
-    // Change the group to 'tty' regardless of what was previously set
-    inner_gid = 5;
+    // Change the group to "tty" regardless of what was previously set
+    struct group *g = getgrnam("tty");
```
Comment (larsrc-google) [inline src/main/tools/linux-sandbox-pid1.cc:237]: A friendly internal linter suggests using getgrnam_r for extra thread safety.

**Comment by larsrc-google** [comment]
@crydell-ericsson This PR is out of date, could you update it?

**Pull request closed without merging**